src/SDKs/Azure.Base/data-plane/Azure.Base/Pipeline/Policies/FixedRetryPolicy.cs

```diff
@@ -37,10 +27,10 @@ protected override bool IsRetriableResponse(HttpPipelineMessage message, int att
                 return false;
             }
 
-            return Array.IndexOf(RetriableCodes, message.Response.Status) >= 0;
+            return !message.ResponseClassifier.IsFatalErrorResponse(message.Response);
```

Comment:
It seems like we now have two abstractions for almost the same thing/question: is something an error?
i.e. we already have an abstraction for this, just tied to retry policy, and your PR asserts that the abstraction is more general (which might be a valid assertion). But then this PR instead of moving the abstraction from retry policy, adds a new abstraction. Would it be possible to simplify here by removing the old abstraction?

Reply:
We still need the "How long do you wait" part. I can rename methods to `CalculateRetryDelay`.